tests/test_read_strategy.py

```diff
@@ -1,15 +1,95 @@
+import os
+
 import pytest
 
-from liccheck.command_line import read_strategy
+from liccheck.command_line import Strategy, NoValidConfigurationInPyprojectToml, read_strategy
 
 
-def test_absent_sections(tmpfile):
-    tmpfh, tmppath = tmpfile
-    tmpfh.write("""
+class TestReadFromConfig:
+    def test_absent_sections_in_config_file(self, tmpfile):
+        tmpfh, tmppath = tmpfile
+        tmpfh.write("""
 [Licenses]
 """)
-    tmpfh.close()
-    strategy = read_strategy(tmppath)
-    assert strategy.AUTHORIZED_LICENSES == []
-    assert strategy.UNAUTHORIZED_LICENSES == []
-    assert strategy.AUTHORIZED_PACKAGES == {}
+        tmpfh.close()
+        strategy = Strategy.from_config(strategy_file=tmppath)
+        assert strategy.AUTHORIZED_LICENSES == []
+        assert strategy.UNAUTHORIZED_LICENSES == []
+        assert strategy.AUTHORIZED_PACKAGES == {}
+
+
+class TestReadFromPyprojectToml:
+    def test_raises_if_no_pyproject_toml(self):
+        with pytest.raises(NoValidConfigurationInPyprojectToml):
+            Strategy.from_pyproject_toml()
+
+    @pytest.mark.usefixtures("empty_pyproject_toml_in_cwd")
+    def test_falls_back_to_strategy_if_no_liccheck_section_in_pyproject_toml(self):
+        with pytest.raises(NoValidConfigurationInPyprojectToml):
+            Strategy.from_pyproject_toml()
+
+    @pytest.mark.usefixtures("pyproject_toml_with_liccheck_section_in_cwd")
+    def test_with_liccheck_section_in_pyproject_toml(self):
+        strategy = Strategy.from_pyproject_toml()
+        assert strategy.AUTHORIZED_LICENSES == [
+            "bsd",
+            "new bsd",
+            "bsd license"
+        ]
+        assert strategy.UNAUTHORIZED_LICENSES == [
+            "gpl v3"
+        ]
+        assert strategy.AUTHORIZED_PACKAGES == {
+            "uuid": "1.30"
+        }
+
+    @pytest.fixture
+    def empty_pyproject_toml_in_cwd(self, tmpdir):
+        cwd = os.getcwd()
+        os.chdir(str(tmpdir))
+        open("pyproject.toml", "w").close()
+        yield
+        os.chdir(cwd)
+
+    @pytest.fixture
+    def pyproject_toml_with_liccheck_section_in_cwd(self, empty_pyproject_toml_in_cwd):
+        with open("pyproject.toml", "w") as file:
+            file.write(
+                """
+                [tool.liccheck]
+                authorized_licenses = [
+                  "BSD",
+                  "new bsd",
+                  "bsd license"
+                ]
+                unauthorized_licenses = [
+                  "gpl v3"
+                ]
+                [tool.liccheck.authorized_packages]
```

Comment:
Why not incuding `authorized_packages` in the `tool.liccheck` section above? Also, does this syntax supports all version requirements operators, i.e. can we specify for example `uuid >= 1.30, < 2`?

Reply:
`authorized_packages` must be a dictionary to map the package name to the allowed versions. And afaik there is no other way to define a dictionary within a dictionary in toml than this (e.g. indendation seems to be ignored entirely). See https://github.com/toml-lang/toml

> Also, does this syntax supports all version requirements operators, i.e. can we specify for example uuid >= 1.30, < 2?

Yes, see the example in the PR description

Reply:
Great! Before merging, could you add a paragraph in the readme regarding this new feature? Thanks a lot.